core/geometry/src/polyface/Polyface.ts

```diff
@@ -316,126 +355,145 @@ export class IndexedPolyface extends Polyface {
     this.data.point.pushXYZ(point.x, point.y, point.z);
     return this.data.point.length - 1;
   }
-
-  /** add a point.
-   * @returns Returns the zero-based index of the added point.
+  /**
+   * Add a point to point array.
+   * @param x the x coordinate of point.
+   * @param y the y coordinate of point.
+   * @param z the z coordinate of point.
+   * @returns the zero-based index of the added point.
    */
-  public addPointXYZ(x: number, y: number, z: number): number { this.data.point.pushXYZ(x, y, z); return this.data.point.length - 1; }
-  /** Add a uv param.
-   * @returns 0-based index of the added param.
+  public addPointXYZ(x: number, y: number, z: number): number {
+    this.data.point.pushXYZ(x, y, z);
+    return this.data.point.length - 1;
+  }
+  /**
+   * Add a uv parameter to the parameter array.
+   * @param param the parameter.
+   * @returns zero-based index of the added param.
    */
   public addParam(param: Point2d): number {
-    if (!this.data.param) this.data.param = new GrowableXYArray();
+    if (!this.data.param)
+      this.data.param = new GrowableXYArray();
     this.data.param.push(param);
     return this.data.param.length - 1;
   }
-  /** Add a uv parameter to the parameter array.
+  /**
+   * Add a uv parameter to the parameter array.
+   * @param u the u part of parameter.
+   * @param v the v part of parameter.
    * @param priorIndexA first index to check for possible duplicate value.
    * @param priorIndexB second index to check for possible duplicate value.
-   * @returns 0-based index of the added or reused param.
+   * @returns zero-based index of the added or duplicate parameter.
    */
   public addParamUV(u: number, v: number, priorIndexA?: number, priorIndexB?: number): number {
-    if (!this.data.param) this.data.param = new GrowableXYArray();
+    if (!this.data.param)
+      this.data.param = new GrowableXYArray();
     if (priorIndexA !== undefined && this.data.isAlmostEqualParamIndexUV(priorIndexA, u, v))
       return priorIndexA;
     if (priorIndexB !== undefined && this.data.isAlmostEqualParamIndexUV(priorIndexB, u, v))
       return priorIndexB;
     this.data.param.pushXY(u, v);
     return this.data.param.length - 1;
   }
-
-  /** Add a normal vector
+  /**
+   * Add (a clone of) a normal vector to the normal array.
+   * @param normal the normal vector.
    * @param priorIndexA first index to check for possible duplicate value.
    * @param priorIndexB second index to check for possible duplicate value.
-   * @returns 0-based index of the added or reused normal.
+   * @returns zero-based index of the added or duplicate normal.
    */
   public addNormal(normal: Vector3d, priorIndexA?: number, priorIndexB?: number): number {
+    // check if `normal` is duplicate of `dataNormal` at index `i`
```

Comment:
I can even remove the first parameter and change the lambda function to
``` 
    const normalIsDuplicate = (i: number) => {
      const distance = this.data.normal!.distanceIndexToPoint(i, normal);
      return distance !== undefined && Geometry.isSmallMetricDistance(distance);
    };
 ```
 and call it like ```normalIsDuplicate(priorIndexA)```